src/pkg/scan/export/filter_processor.go

```diff
@@ -0,0 +1,127 @@
+package export
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/common"
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/lib/selector"
+	"github.com/goharbor/harbor/src/lib/selector/selectors/doublestar"
+	"github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/pkg/project/models"
+	"github.com/goharbor/harbor/src/pkg/repository"
+	"github.com/goharbor/harbor/src/pkg/tag"
+	"github.com/goharbor/harbor/src/pkg/user"
+)
+
+type FilterProcessor interface {
+	ProcessProjectFilter(ctx context.Context, userName string, projectsToFilter []int64) ([]int64, error)
+	ProcessRepositoryFilter(ctx context.Context, filter string, projectIds []int64) ([]*selector.Candidate, error)
+	ProcessTagFilter(ctx context.Context, filter string, repositoryIds []int64) ([]*selector.Candidate, error)
+}
+
+type DefaultFilterProcessor struct {
+	repoMgr    repository.Manager
+	tagMgr     tag.Manager
+	usrMgr     user.Manager
+	projectMgr project.Manager
+}
+
+// NewFilterProcessor constructs an instance of a FilterProcessor
+func NewFilterProcessor() FilterProcessor {
+	return &DefaultFilterProcessor{
+		repoMgr:    repository.New(),
+		tagMgr:     tag.Mgr,
+		usrMgr:     user.Mgr,
+		projectMgr: project.New(),
+	}
+}
+
+func (dfp *DefaultFilterProcessor) ProcessProjectFilter(ctx context.Context, userName string, projectIdsToFilter []int64) ([]int64, error) {
+	// get the user id of the current user
+
+	usr, err := dfp.usrMgr.GetByName(ctx, userName)
+	if err != nil {
+		return nil, err
+	}
+	logger.Infof("Retrieved user id :%d for user name : %s", usr.UserID, userName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	projects, err := dfp.projectMgr.List(ctx, q.New(q.KeyWords{"member": &models.MemberQuery{UserID: usr.UserID, Role: common.RoleProjectAdmin}}))
+
+	if err != nil {
+		return nil, err
+	}
+	logger.Infof("Selected %d projects administered by user %s ", len(projects), userName)
+	projectIds := make([]int64, 0)
+	for _, proj := range projects {
+		projectIds = append(projectIds, proj.ProjectID)
+	}
+
+	// check if the project ids specified in the filter are present in the list
+	// of projects of which the current user is a project admin
+	if projectIdsToFilter == nil || len(projectIdsToFilter) == 0 {
+		return projectIds, nil
+	}
+	m := make(map[int64]bool)
+	for _, projectId := range projectIds {
+		m[projectId] = true
+	}
+	filtered := make([]int64, 0)
+
+	for _, filteredProjId := range projectIdsToFilter {
+		if _, ok := m[filteredProjId]; ok {
```

Comment:
as the map value is boolean type, so here can just use `if m[filteredProjId] {...}`

Reply:
Done